### DISCUSSION TITLE
Can we define a block without neurons?

### DISCUSSION THREAD

**Discussion by fangwei123456** [lava-nc/lava-dl#81] [Q&A] [answered]
In deep SNNs, we may use the structure like 'Conv-Neuron-Pool-Conv-Neuron'. In such a case, we use the pooling layer without spiking neurons. But I find that in [lava.lib.dl.slayer.block.cuba.Pool](https://lava-nc.org/lava-lib-dl/slayer/block/block.html#lava.lib.dl.slayer.block.cuba.Pool), we also need to include the spiking neurons in pooling layer. So, my question is can we define a block without neurons?

**Comment by bamsumit** [comment, marked as answer]
`slayer.block` is a convenient method that combines synapses and neurons. Typically in neuromorphic hardware, you need to have neurons. That is why the block interface provides both neuron and synapse combined. However, if you need only the synapse, you can directly call `slayer.synapse.Pool` and it should work.

**Reply by fangwei123456** [reply]
Thanks for your reply! Another question is how can I convert `lava.lib.dl.slayer.synapse.layer.Pool` to `Processes` in lava? I find that [Supported layer types](https://lava-nc.org/lava-lib-dl/netx/netx.html#supported-layer-types) does not contain a pool without neuron.

**Reply by bamsumit** [reply]
You are right. NetX network description does not support such structure nor the Loihi hardware. Therefore, we have no plans of supporting such a structure. If you really want to do it, you can break the network into two NetX halves and connect them with a custom lava process for pooling.

**Reply by fangwei123456** [reply]
Thanks!
